src/components/tl/ucp/tl_ucp_service_coll.c

```diff
@@ -52,15 +93,23 @@ ucc_status_t ucc_tl_ucp_service_allreduce(ucc_base_team_t *team, void *sbuf,
     if (status != UCC_OK) {
         goto free_task;
     }
+
+    status = ucc_tl_ucp_service_coll_start_executor(&task->super);
```

Comment:
How is this executor stopped? Are you setting EXECUTOR_STOP flag on the task somewhere? How it is finalized?

Why don't you start executor for the allgather below?

Reply:
right, added executor_stop flag